lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -101,62 +115,128 @@ bool SessionStateEnforcer::m5g_init_session_credit(
   } else {
     session_map[imsi].push_back(std::move(session_state));
   }
-  MLOG(MINFO) << "Added a session (" << session_map[imsi].size()
-              << ") for IMSI " << imsi << " with session context ID ";
+  MLOG(MDEBUG) << "Added a session (" << session_map[imsi].size()
+               << ") for IMSI " << imsi << " with session context ID ";
   return true;
 }
 
-void SessionStateEnforcer::handle_session_init_rule_updates(
+bool SessionStateEnforcer::m5g_update_session_context(
     SessionMap& session_map, const std::string& imsi,
-    SessionState& session_state) {
-  uint32_t l_teid = 0;
-  auto itp        = pdr_map_.equal_range(imsi);
+    std::unique_ptr<SessionState>& session_state,
+    SessionUpdate& session_update) {
+  uint32_t gnode_teid = 0;
+  bool gnb_teid_get   = true;
+  bool upf_teid_get   = false;
+
+  /* Check and update latest session rules
+   * we get gnodeb TEID, and IP address details here
+   */
+  auto session_id                       = session_state->get_session_id();
+  SessionStateUpdateCriteria session_uc = session_update[imsi][session_id];
+  gnode_teid                            = update_session_rules(
+      imsi, session_state, session_uc, gnb_teid_get, upf_teid_get);
+  if (!gnode_teid) return false;

```

Comment:
here and everywhere else, but I tend to prefer
```
if (!gnode_teid) {
   return false;
}
```

Reply:
Yes, Fixed it.